SPECIFICATION.md

```diff
@@ -122,19 +122,19 @@ If we had received the hash for tree index 5 before and verified it then we woul
 
-As an optimization we can use the remote's "have" messages (see the "Wire Protocol" section) to figure out which hashes it already has. For example if the remote has block #3 then we do not need to send any hashes since it will already have verified hash 6 and 5.
+As an optimization we can use the remote's "have" messages (see the ["Wire Protocol" section](#messageshave)) to figure out which hashes it already has. For example if the remote has block #3 then we do not need to send any hashes since it will already have verified hash 6 and 5.
 
 Hyperdrive's content integrity approach is similar to parts of [ppspp](https://tools.ietf.org/html/rfc7574) and additional information can found in that specification.
 
 ## Deduplication
 
-Assuming you have two different feeds that are sharing similar data we want to able to fetch as little as possible duplicate data. The merkle tree structure helps us achieve this by content addressing the feeds.
+Assuming you have two different feeds that are sharing similar data, we want to minimize fetching of duplicate data. The merkle tree structure helps us achieve this by content addressing the feeds.
 
-For example if we were to produce the exact same feed on two different computers using the technique described in the above section the feeds would end up with the same root tree hashes and the same feed id and it would therefore be the same feed.
+For example if we were to produce the exact same feed on two different computers using the technique described in the above section, the feeds would end up with the same root tree hashes and the same feed id. The two feeds would therefore be the same feed.
 
-A more interesting case is sharing two similar feeds that are not 100% the same but share partial sections. This could be two different versions of the same file, an old one, and an updated one with a few changes. In this case feed ids would be different. However since every block is being delivered with the tree hashes necessary to verify it against the root of the merkle tree we can maintain a simple index that points from the parent hashes in the merkle trees to the data blocks they represent and check against this index to see if we have already fetched other parts a feed before shared by another feed.
+A more interesting case is sharing two similar feeds that are not 100% the same but share partial sections. This could be two different versions of the same file, an old one, and an updated one with a few changes. In this case, feed ids would be different. However, since every block is being delivered with the tree hashes necessary to verify it against the root of the merkle tree, we can maintain a simple index that points from the parent hashes in the merkle trees to the data blocks they represent and check against this index to see if we have already fetched other parts a feed before shared by another feed.
```

Comment:
>  In this case, feed ids would be different. However, since every block is being delivered with the tree hashes necessary to verify it against the root of the merkle tree, we can maintain a simple index that points from the parent hashes in the merkle trees to the data blocks they represent and check against this index to see if we have already fetched other parts a feed before shared by another feed.

This is kind of a run on sentence thats hard to understand whats going on.  Maybe provide (a link to some code?) an example of how to create and check against said index?